pinot-core/src/main/java/org/apache/pinot/core/segment/processing/mapper/SegmentMapper.java

```diff
@@ -180,13 +188,11 @@ private boolean completeMapAndTransformRow(RecordReader recordReader, GenericRow
         reuse = recordReader.next(reuse);
         _recordEnricherPipeline.run(reuse);
 
-        if (reuse.getValue(GenericRow.MULTIPLE_RECORDS_KEY) != null) {
-          //noinspection unchecked
-          for (GenericRow row : (Collection<GenericRow>) reuse.getValue(GenericRow.MULTIPLE_RECORDS_KEY)) {
-            transformAndWrite(row);
+        _transformPipeline.processRow(reuse, reusedResult);
```

Comment:
Is processRow is exactly the same as transformAndWrite ?

Reply:
Yes it does the same thing as transformAndWrite minus the `writeRecord` part which is called separately below.
Actually `IngestionUtils.shouldIngestRow(transformedRow)` check is also applied in `processRow` while building the `reusedResult` so I'll remove that check from here as well.